src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CKeyStore.java

```diff
@@ -234,11 +250,10 @@ public void delete() throws KeyStoreException {
 
     /*
      * The keystore location.
-     * Case is not significant.
      */
-    private final String storeLocation;
+    private final KeyStoreLocation storeLocation;
 
-    CKeyStore(String storeName, String storeLocation) {
+    CKeyStore(String storeName, KeyStoreLocation storeLocation) {
```

Comment:
Why not just an `int` here? The creation of a separate class `keyStoreLocation` seems not necessary. If you want code to be readable, just add `public static final int CURRENTUSER = 0`, etc.

Reply:
I was using type safety to remove the chance of non-expected values being passed to the C function.  Implemented your recommendation as its a simple contract between two files